docs/source/CartesianConfig.rst

```diff
@@ -363,6 +363,135 @@ statements addressing all objects. Note: This is contrary to the way the
+============================
+
+The ‘``join``’ filter combined with the ‘``suffix``’ operator can be utilized
+together in order to produce guest variants with different guest OS or other
+types of configuration within the same test. Adding the ‘``suffix``’ keyword
+at the end of each variant to be combined and joining all the variants in the
+end will produce one final variant product with all participating variant
+dictionaries separately suffixed and combined into one final dictionary. For
+instance the following definition
+
+::
+
+    variants:
+        - one:
+            key1 = Hello
+            key2 = Brave
+            suffix _v1
+        - two:
+            key1 = Bye
+            key2 = Brave
+            key3 = World
+            suffix _v2
+        - three:
+    variants:
+        - four:
+            key4 = foo
+            only one
+        - five:
+            key4 = bar
+        - six:
+            key1 = foo
+            key2 = bar
+            key3 = baz
+            only two
+
+    join one two
+
+will join variants ``one`` and ``two`` with suffixed parameters
+``key1_v1 = Hello`` and ``key1_v2 = Bye`` in each of the following resulting
+dictionaries:
+
+::
+
+    dict    1:  four.one.five.two
+        dep = []
+        key1_v1 = Hello
+        key1_v2 = Bye
+        key2 = Brave
+        key3 = World
+        key4 = bar
+        name = four.one.five.two
+        shortname = four.one.five.two
+    dict    2:  four.one.six.two
+        dep = []
+        key1 = foo
+        key1_v1 = Hello
+        key1_v2 = Bye
+        key2 = bar
+        key2_v1 = Brave
+        key2_v2 = Brave
+        key3 = baz
+        key3_v2 = World
+        key4 = foo
+        name = four.one.six.two
+        shortname = four.one.six.two
+    dict    3:  five.one.two
+        dep = []
+        key1_v1 = Hello
+        key1_v2 = Bye
+        key2 = Brave
+        key3 = World
+        key4 = bar
+        name = five.one.two
+        shortname = five.one.two
+    dict    4:  five.one.six.two
+        dep = []
+        key1 = foo
+        key1_v1 = Hello
+        key1_v2 = Bye
+        key2 = bar
+        key2_v1 = Brave
+        key2_v2 = Brave
+        key3 = baz
+        key3_v2 = World
+        key4 = bar
+        name = five.one.six.two
+        shortname = five.one.six.two
+
+Regarding the choice of variants, using the ``join one two`` also plays the
+role of a filter for ``three`` which restricts the final selection to all
+joined ``one``-``two`` test variants. The ``join one`` operation is thus
+equivalent to an ``only one`` filter. The outcomes above have additional
+filters thus restricting the initial joined Cartesian products
+
+::
+
+    dict    1:  four.one.two
+    dict    2:  four.one.five.two
+    dict    3:  four.one.six.two
+    dict    4:  five.one.four.two
+    dict    5:  five.one.two
+    dict    6:  five.one.six.two
+    dict    7:  six.one.four.two
+    dict    8:  six.one.five.two
+    dict    9:  six.one.two
+
```

Comment:
Hi @pevogam  can you explain me how did you obtain above dictionaries? It is not clear to me from text. Thanks

Reply:
These are the initial joined Cartesian products. As I said there, the ``join one two`` is equivalent to a filter of the joined `one..two` variants excluding variant `three` and just obtaining all `one` variants joined with all `two` variants.

Reply:
Sorry @pevogam I still do not follow how did you obtain those dictionaries. What is the input cfg file?

Reply:
As the sentence including this list says, these are `the initial joined Cartesian products` without filters applied so it is the same config file without the `only` restrictions. The entire sentence reads that adding the filters (as in the sample input config file) restricts this initial list of 9 variants into the one of 4 variants. I added this to include some of the comments of @ldoktor about the different lists he obtains but I am not sure if it makes things less clear. Of course, your suggestions on how to restate this are welcome if there are some and the sentence is still unclear.

Reply:
I understand now what you are trying to say @pevogam. But I would reformulate your last sentence to:

```
The outcome above are additionally filtered by `only` filter, which restricts the initial joined Cartesian products:
```
IMHO it is better understandable.
Since outcome does not have any filter but is filtered. Filter `only` restricts the initial joined Cartesian products.

Reply:
Hi @rdkdd, acknowledged. I have just force pushed this change.